modal/functions.py

```diff
@@ -960,6 +960,14 @@ def get_build_def(self) -> str:
         """mdmd:hidden"""
         return f"{inspect.getsource(self._raw_f)}\n{repr(self._build_args)}"
 
+    def get_globals(self):
+        from cloudpickle.cloudpickle import _extract_code_globals
+
+        func = self._raw_f
+        f_globals_ref = _extract_code_globals(func.__code__)
```

Comment:
Is this basically the same as inspect.getclosurevars?

Minor preference for putting this in function_utils.py maybe?

Reply:
`inspect.getclosurevars` has some weird behavior, e.g. see this issue https://github.com/python/cpython/issues/80878:

> We are taking names from func.__code__.co_names, which also contains names of object's attributes we visited. As a result, we may get more names unexpectedly in 'globals', 'builtins', and even 'unbound'.

Makes sense to move it to function_utils.py, moved